core/src/main/java/io/grpc/internal/ClientCallImpl.java

```diff
@@ -129,9 +139,20 @@ public void cancelled(Context context) {
     return this;
   }
 
+  ClientCallImpl<ReqT, RespT> setCompressorRegistry(CompressorRegistry compressorRegistry) {
+    this.compressorRegistry = compressorRegistry;
+    return this;
+  }
+
+  ClientCallImpl<ReqT, RespT> setKnownMessageEncodings(Set<String> knownMessageEncodings) {
```

Comment:
How about the word share?  shareKnownMessage....   

That way its clear who the actual owner of the data is, but indicates that callees can mutate it?